ssg/templates.py

```diff
@@ -8,50 +8,62 @@
-template_type = ssg.utils.enum("remediation", "check")
-
-languages = {
-    "anaconda": templating_lang("anaconda", ".anaconda", template_type.remediation, "anaconda"),
-    "ansible": templating_lang("ansible", ".yml", template_type.remediation, "ansible"),
-    "bash": templating_lang("bash", ".sh", template_type.remediation, "bash"),
-    "blueprint": templating_lang("blueprint", ".toml", template_type.remediation, "blueprint"),
-    "ignition": templating_lang("ignition", ".yml", template_type.remediation, "ignition"),
-    "kubernetes": templating_lang("kubernetes", ".yml", template_type.remediation, "kubernetes"),
-    "oval": templating_lang("oval", ".xml", template_type.check, "oval"),
-    "puppet": templating_lang("puppet", ".pp", template_type.remediation, "puppet"),
-    "sce-bash": templating_lang("sce-bash", ".sh", template_type.remediation, "sce")
+
+TemplateType = ssg.utils.enum("REMEDIATION", "CHECK")
+
+LANGUAGES = {
+    "anaconda": TemplatingLang("anaconda", ".anaconda", TemplateType.REMEDIATION, "anaconda"),
+    "ansible": TemplatingLang("ansible", ".yml",        TemplateType.REMEDIATION, "ansible"),
+    "bash": TemplatingLang("bash", ".sh",               TemplateType.REMEDIATION, "bash"),
+    "blueprint": TemplatingLang("blueprint", ".toml",   TemplateType.REMEDIATION, "blueprint"),
+    "ignition": TemplatingLang("ignition", ".yml",      TemplateType.REMEDIATION, "ignition"),
+    "kubernetes": TemplatingLang("kubernetes", ".yml",  TemplateType.REMEDIATION, "kubernetes"),
+    "oval": TemplatingLang("oval", ".xml",              TemplateType.CHECK,       "oval"),
+    "puppet": TemplatingLang("puppet", ".pp",           TemplateType.REMEDIATION, "puppet"),
+    "sce-bash": TemplatingLang("sce-bash", ".sh",       TemplateType.REMEDIATION, "sce")
 }
-preprocessing_file_name = "template.py"
 
-templates = dict()
+PREPROCESSING_FILE_NAME = "template.py"
+TEMPLATE_YAML_FILE_NAME = "template.yml"
 
 
-class Template():
-    def __init__(self, template_root_directory, name):
-        self.template_root_directory = template_root_directory
+class Template:
+    def __init__(self, templates_root_directory, name):
+        self.langs = []
+        self.templates_root_directory = templates_root_directory
         self.name = name
-        self.template_path = os.path.join(self.template_root_directory, self.name)
-        self.template_yaml_path = os.path.join(self.template_path, "template.yml")
-        self.preprocessing_file_path = os.path.join(self.template_path, preprocessing_file_name)
-
-    def load(self):
+        self.template_path = os.path.join(self.templates_root_directory, self.name)
+        self.template_yaml_path = os.path.join(self.template_path, TEMPLATE_YAML_FILE_NAME)
+        self.preprocessing_file_path = os.path.join(self.template_path, PREPROCESSING_FILE_NAME)
+
+    @classmethod
+    def load_template(cls, template_root_directory, name):
+        maybe_template = cls(template_root_directory, name)
```

Comment:
Inconsistent nameing variables - here it's `template_root_directory` but in the constructor it's `templates_root_directory` with "s" so I think here it should also be  `templates_root_directory` with "s".

Reply:
Ah, missed this one.

Reply:
Fixed.